resource/evaluators/expr_eval_api.cpp

```diff
@@ -315,6 +365,34 @@ int expr_eval_api_t::evaluate (const std::string &e,
     return rc;
 }
 
+int expr_eval_api_t::extract (const std::string &e,
+                              const expr_eval_target_base_t &target,
+                              std::vector<std::pair<std::string, std::string>> &predicates)
+{
+    int rc = -1;
+    pred_op_t op;
+    std::size_t next, at;
+
+    if ((rc = extract_paren (e, target, 0, next, predicates)) < 0)
+        goto done;
+    at = next;
+
+    while (at <= e.find_last_not_of (" \t")) {
```

Comment:
Since `e` is static, may as well save the search each time.

```suggestion
    auto last_before_space = e.find_last_not_of (" \t");
    while (at <= last_before_space) {
```

Reply:
Good idea. I've added a separate commit to apply the change to the existing functions, too.